src/quo2/components/qr_code/qr.cljs

```diff
@@ -0,0 +1,32 @@
+(ns quo2.components.qr-code.qr
+  (:require [react-native.core :as rn]))
+
+(defn user-profile-qr-code
+  [{:keys [key-uid public-key port qr-size]}]
+  (let [profile-qr-url         (str "https://join.status.im/u/" public-key)
+        base-64-qr-url         (js/btoa profile-qr-url)
+        profile-image-type     "large"
+        error-correction-level 4
+        superimpose-profile?   true
+        media-server-url       (str "https://localhost:"
+                                    port
+                                    "/GenerateQRCode?level="
```

Comment:
Apart from the important feedback @flexsurfer shared about extracting this code outside `quo2`, it's also a real shame RN doesn't fully support the URL spec (it's buggy and the recommendation is to use a polyfill library) or resort to concatenating strings as do we here. It's the first time I see a development environment without proper support for building a URL :disappointed: https://github.com/facebook/react-native/issues/16434#issuecomment-922365123